### PR TITLE
Update to match latest Java library (2021) behavior / Adds initial test for S2RegionCoverer

### PR DESCRIPTION
**Pull request by liang3404814**
This PR mirrors https://github.com/google/s2-geometry-library-java/commit/8525c2d2f57de66272eb8358fdee0df20084acf7 to get calculations up to date, and adds tests for S2RegionCoverer. Note this version contains breaking changes (removing outdated projection methods, getter/function changes, etc).

## Background

Google released https://github.com/google/s2-geometry-library-java/commit/8525c2d2f57de66272eb8358fdee0df20084acf7, containing "9 years of internal changes". This made some significant tweaks to how the cells/projections are calculated and defined, among with a huge number of cleanups / creating new classes. Based on the definition of `avgArea`, this is the state of all other existing libraries that mirror Google's implementation:

Swift: PI / 6 (Old)
(https://github.com/philip-bui/s2-geometry-swift/blob/d2ba247f712ab348eaced369f0c5094120d27206/Sources/Classes/S2Projections.swift#L104)
Kotlin: 4 x PI / 6 (New)
([https://github.com/Enovea/s2-geometry-kotlin/blob/ca92756012efccf54324e55b1d0a3c10[…]58e4/src/main/kotlin/dilivia/s2/coords/S2QuadraticProjection.kt](https://github.com/Enovea/s2-geometry-kotlin/blob/ca92756012efccf54324e55b1d0a3c102ea958e4/src/main/kotlin/dilivia/s2/coords/S2QuadraticProjection.kt#L64))
C++: 4 x PI / 6 (New)
(https://github.com/google/s2geometry/blob/0c4c460bdfe696da303641771f9def900b3e440f/src/s2/s2metrics.cc#L110)
Java: 4 x PI / 6 (New)
([https://github.com/google/s2-geometry-library-java/blob/2ebcfda1f1b5e1d417b588a59a[…]2944cb99c28ed/src/com/google/common/geometry/S2Projections.java](https://github.com/google/s2-geometry-library-java/blob/2ebcfda1f1b5e1d417b588a59a12944cb99c28ed/src/com/google/common/geometry/S2Projections.java#L601))
TS (existing): PI / 6 (Old)
(https://github.com/vekexasia/nodes2-ts/blob/master/src/S2Projections.ts#L78)

It can be seen from the above that most libraries are using Google's updated methods, so upgrading should bring us closer to other libraries' behaviors. Doing many of these changes also makes this library more ready to port the rest of classes in the future.

## Note on S2RegionCoverer

I generated tests from Java but only using them as a starting point to verify the covering cells are in fact, covering the region. It isn't a good idea to use a parity test (with Java's) for the `getCovering` method because it uses a PriorityQueue, which doesn't have a well-agreed-on tie-breaking method when it comes to two cells that have the same priority, thus the method can return results that are different from Java's but equally valid in nature.

### CODE REVIEW

File: src/utils/comparison.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export function compareDouble(x: number, y: number) {
+  // const diff = Math.abs(x - y);
```
Comment (vekexasia) [inline src/utils/comparison.ts:2]: Can you remove the comments here? or is the implementtio within the comments the correct version?

**Review by vekexasia — CHANGES_REQUESTED** [review]

**Comment by vekexasia** [comment]
Hello @liang3404814 I did some modifications today to the repo so I took the liberty to merge my latest commits into this branch since there were some conflicts.

File: src/utils/comparison.ts (hunk 1 above)
Comment (liang3404814) [reply src/utils/comparison.ts:2]: Yes! I forgot to clean this up---this epsilon check was breaking some tests so I changed it to the more sensitive direct comparison.

**Review by vekexasia — APPROVED** [review]
thanks :)

**Pull request merged**